frontend/src/static/js/v_authorship.js

```diff
@@ -42,32 +42,74 @@ window.vAuthorship = {
       toReverseSortFiles: true,
       hasActiveFile: true,
       filterSearch: '*',
+      searchBarValue: '',
       sortingFunction: (a, b) => -1 * window.comparator(filesSortDict.lineOfCode)(a, b),
     };
   },
 
   watch: {
     filesSortType() {
       this.sortFiles();
+      window.addHash('authorshipSortBy', this.filesSortType);
+      window.encodeHash();
     },
     toReverseSortFiles() {
       this.sortFiles();
+      window.addHash('reverseAuthorshipOrder', this.toReverseSortFiles);
+      window.encodeHash();
+    },
+    selectedFileTypes() {
+      if (this.filterType === 'checkboxes') {
+        this.updateFileTypeHash();
+      }
     },
     selectedFiles() {
       setTimeout(this.updateCount, 0);
     },
     filterType() {
       if (this.filterType === 'checkboxes') {
-        const searchBar = document.getElementById('search');
-        searchBar.value = '';
+        this.searchBarValue = '';
         this.filterSearch = '*';
+        this.updateFileTypeHash();
       } else {
         this.selectedFileTypes = this.fileTypes.slice();
+        this.updateGlobHash();
       }
     },
   },
 
   methods: {
+    retrieveHashes() {
+      window.decodeHash();
+
+      const hash = window.hashParams;
+
+      switch (hash.authorshipSortBy) {
+      case 'path':
+      case 'fileName':
+      case 'fileType':
+        this.filesSortType = hash.authorshipSortBy;
+        break;
+      default:
+        // Invalid value, use the default value
+      }
+
+      if (hash.reverseAuthorshipOrder === 'false') {
+        this.toReverseSortFiles = false;
+      }
+
+      if (hash.authorshipFilesGlob) {
+        this.indicateSearchBar();
+        this.searchBarValue = hash.authorshipFilesGlob === '*'
```

Comment:
A new variable is needed as the dom might not have mounted when `retrieveHashes` is executed.
In such a case, `document.getElementById('search')` will fail.

Reply:
I dont really get the reason for having this new variable? Can just set filterSearch = '' in data and bind it to the search bar, then in `selectedFiles` do sth like `minimatch(file.path, this.filterSearch || '*'`

Reply:
Good idea, I think I'll make `this.filterSearch || '*'` a computed property, thanks!